Improvements to the `sda-admin` script

andkaha opened the pull request:
This PR addresses the issues in #37  

This PR closes #35
This PR closes #37

andkaha committed:
Remove use of `--json` with `curl`

This was not compatible with the `curl` available on a deployment
system.

andkaha committed:
Do not add missing `.c4gh` filename suffix to filenames

This makes it possibly to work with files that were uploaded with other
filenames suffixes than `.c4gh`.

The `upload` functionality will still encrypt file and these files will
get the `.c4gh` filename suffix.

andkaha committed:
Add `-a` option to some sub-commands

The `ingest`, `accession`, and `dataset` sub-commands gain a `-a`
command line option.  This option is valid when the sub-command is used
for listing the available filenames in a given queue, and makes the
command list the filenames in _all_ buckets.

Note that the output format changes when `-a` is used as the filenames
are prefixed with the bucket thathey reside in.

andkaha committed:
Add `--user` as an alias for `--s3-access-key`

andkaha committed:
Replace any `@` in the S3 access key with underscore

andkaha committed:
Don't allow upload targed directory path to start with `./`

There has been a change in `sda-cli` or in MinIO such that if we
specify the a target directory with `-targetDir`, that directory
path may not contain a dot component.  We previously used a dot
at the start (as in `./some/dir/path`) but this commit removes
any `./` substring from the start of the path.

andkaha committed:
Ensure that the upload target path is valid

aaperis commented on scripts/sda-admin:
```diff
@@ -21,8 +21,9 @@ SDA_KEY=${SDA_KEY-crypt4gh_key.pub}			# --sda-key pathname
-# There is also S3_ACCESS_KEY (--s3-access-key), but since its value
-# depend on the final value of "$SDA_CONFIG", we can't set it here.
+# There is also S3_ACCESS_KEY (--s3-access-key and its alias, --user),
+# but since its value depend on the final value of "$SDA_CONFIG", we
```
```suggestion
# but since its value depends on the final value of "$SDA_CONFIG", we
```

aaperis reviewed (commented):
This looks good! I made a few tests and I have some notes:
1. the script does not return any error message if I provide a wrong path to a file that I want to upload, i.e. if the file doesnot exist
2. I am skeptical about the `""` option for `sda-admin accession` and `sda-admin dataset` since this could lead to unwanted situations. 
Consider for example that more than one user have uploaded files and then the admin accidentally does a mass accession or dataset assignment using `""`. So, I would suggest that we keep the `""` option only for the `ingest` subcommand.

kostas-kou reviewed (approved):
Looks very good and I have tested it a bit around. I have just a minor comment which is maybe not so relevant with this PR but it would be nice if there was the `-k` option for insecure connections in the curl command. I am making this comment because I am using this script for the Big Picture and it is not working without the insecure connection.

andkaha commented:
> Looks very good and I have tested it a bit around. I have just a minor comment which is maybe not so relevant with this PR but it would be nice if there was the `-k` option for insecure connections in the curl command. I am making this comment because I am using this script for the Big Picture and it is not working without the insecure connection.

One solution for this that doesn't require that you change the script:
```
echo '--insecure' >.curlrc
export CURL_HOME=$PWD
```

Basically, this creates a Curl config file in the current directory containing the `--insecure` option. By setting `CURL_HOME` to the path of the current directory, you make all invocations of `curl` read that file first.

Commit:
Update scripts/sda-admin

Co-authored-by: Alex Aperis <76202622+aaperis@users.noreply.github.com>

kostas-kou commented:
> export CURL_HOME=$PWD

It works like a charm. Thanks Andreas

andkaha committed:
Address comments from Alex

- Don't allow empty filenames with `dataset` subcommand.

    - Addresses comment from Alex that allowing `""` as an argument to
`dataset` may cause headaches.

- Inform the user of not finding the files that they want to upload.

    - In the `encrypt` function, instead of _skipping_ files that aren't
found, give up and return failure to the calling function.

andkaha commented:
> This looks good! I made a few tests and I have some notes:
> 
> 1. the script does not return any error message if I provide a wrong path to a file that I want to upload, i.e. if the file doesnot exist
> 2. I am skeptical about the `""` option for `sda-admin accession` and `sda-admin dataset` since this could lead to unwanted situations.
>    Consider for example that more than one user have uploaded files and then the admin accidentally does a mass accession or dataset assignment using `""`. So, I would suggest that we keep the `""` option only for the `ingest` subcommand.

These comments hopefully addressed with the most recent commit.

aaperis reviewed (approved):
Looks great!
